### PR TITLE
Remove Stackdriver, Elasticsearch content from Tasks/Monitoring.. and Concepts/Cluster Admin/Logging Arch

### PR DESCRIPTION
**Pull request by aimeeu**
- Remove Stackdriver, Elasticsearch, GCE content from Tasks/Monitoring, Logging, and Debugging section because Kubernetes doesn't specify a logging agent, and no logging agents are
packaged with the Kubernetes core release.
- Refactor existing content: rename [Events in Stackdriver](https://kubernetes.io/docs/tasks/debug-application-cluster/events-stackdriver/) to Events, remove vendor-specific content,  and move the page to the Concepts section.
- Remove references to Stackdriver and Elasticsearch from [Concepts/Cluster Administration/Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/)
- Add 'Events' to glossary.
- Remove images no longer used.

Partial #15748 

Changed content to preview:
- [Tasks](https://deploy-preview-16203--kubernetes-io-master-staging.netlify.com/docs/tasks/) to see that Stackdriver, Elasticsearch pages have been removed from Monitoring, Logging, and Debugging section
- New [Events glossary entry](https://deploy-preview-16203--kubernetes-io-master-staging.netlify.com/docs/reference/glossary/?fundamental=true#term-k8s-events)
- [Concepts/Cluster Administration/Logging Architecture](https://deploy-preview-16203--kubernetes-io-master-staging.netlify.com/docs/concepts/cluster-administration/logging/) to see that Stackdriver and Elasticsearch content has been removed

Note: both FR and IT localizations link to pages that I am removing with this PR
- FR page website/content/fr/docs/concepts/cluster-administration/logging.md
- IT page website/content/it/docs/concepts/cluster-administration/logging.md

### CODE REVIEW

**Comment by aimeeu** [comment]
Copying French and Italian admins
/cc @remyleone 
/cc @micheleberardi

**Comment by remyleone** [comment]
I'm not sure to understand this PR. Could you split your PR to ease up the review?

**Comment by aimeeu** [comment]
@remyleone   Oh, I wasn't expecting you to review this PR. Sorry for the misunderstanding. I wanted to make you aware that the French content links to English pages that are being removed as part of this PR.
website/content/fr/docs/concepts/cluster-administration/logging.md  lines 204-208 
```
Kubernetes ne définit pas d'agent de journalisation, mais deux agents de
journalisation optionnels sont fournis avec la version de Kubernetes :
[Stackdriver (EN)](/docs/user-guide/logging/stackdriver) pour utiliser sur
Google Cloud Platform, et [Elasticsearch
(EN)](/docs/user-guide/logging/elasticsearch). 
```
The English _stackdriver_ and _elasticsearch_ pages are being removed, so the French content will have broken links after this PR merges.

> I'm not sure to understand this PR. Could you split your PR to ease up the review?

**Comment by aimeeu** [comment]
@micheleberardi   I didn't mean to assign you as a Reviewer. Sorry for the mistake. I wanted to make you aware that the Italian content links to English pages that are being removed as part of this PR.
[website/content/it/docs/concepts/cluster-administration/logging.md](https://github.com/kubernetes/website/blob/master/content/it/docs/concepts/cluster-administration/logging.md)  line 132 
```
Kubernetes non specifica un agente di registrazione,
 ma due agenti di registrazione facoltativi sono impacchettati con la versione di Kubernetes: 
[Stackdriver Logging](/docs/user-guide/logging/stackdriver) da utilizzare 
con Google Cloud Platform e [Elasticsearch](/docs/user-guide/logging/elasticsearch). 
```
The English _stackdriver_ and _elasticsearch_ pages are being removed, so the Italian content will have broken links after this PR merges.

**Comment by thecrudge** [comment]
Hi @aimeeu ,

Both Fluentd and Elastic are part of the CNCF. Other than that

/lgtm

File: content/en/docs/reference/glossary/events.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Events
+id: k8s-events
+date: 2019-08-30
+full_link: /docs/concepts/cluster-administration/events/
+short_description: >
+  Events are objects that provide insight into what is happening inside a {{< glossary_tooltip text="cluster" term_id="cluster" >}}.
+
+aka: 
+tags:
```
Comment (sftim) [inline content/en/docs/reference/glossary/events.md]: Also `core-object`?

File: content/en/docs/reference/glossary/events.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+- architecture
+- fundamental
+---
+ Kubernetes events are objects that provide insight into what is happening inside a {{< glossary_tooltip text="cluster " term_id="cluster" >}}, such as what decisions were made by the scheduler or why some pods were evicted from the node.
```
Comment (sftim) [inline content/en/docs/reference/glossary/events.md]:
```suggestion
In Kubernetes, Events are objects that provide insight into what is happening inside a {{< glossary_tooltip text="cluster " term_id="cluster" >}}, such as what decisions were made by the scheduler or why some Pods were evicted from a Node.
```
(see [Use camel case for API objects](https://kubernetes.io/docs/contribute/style/style-guide/#use-camel-case-for-api-objects))

File: content/en/docs/reference/glossary/events.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    
+<!--more--> 
+
+ You can read more about using events
```
Comment (sftim) [inline content/en/docs/reference/glossary/events.md]:
```suggestion
 You can read more about using Events
```

File: content/en/docs/concepts/cluster-administration/events.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+reviewers:
+- piosz
+- x13n
+content_template: templates/concept
+title: Events in Kubernetes
+---
+
+{{% capture overview %}}
+
+Kubernetes events are objects that provide insight into what is happening
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/events.md]: Consider using a glossary definition shortcode here, that automatically includes either the short or full version of the glossary text for this object.

File: content/en/docs/concepts/cluster-administration/events.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+inside a cluster, such as what decisions were made by scheduler or why some
+pods were evicted from the node. 
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+Since events are API objects, they are stored in the {{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} on master. To
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/events.md]:
```suggestion
Since Events are API objects, they are stored in the {{< glossary_tooltip text="apiserver" term_id="kube-apiserver" >}} on the control plane. To
```

File: content/en/docs/concepts/cluster-administration/events.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+avoid filling up master's disk, a retention policy is enforced: events are
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/events.md]:
```suggestion
avoid filling up storage, a retention policy is enforced: events are
```
(there might be more than one server in the control plane, and it might not use its local disk for persistence)

File: content/en/docs/concepts/cluster-administration/events.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+removed one hour after the last occurrence. To provide longer history
+and aggregation capabilities, a third-party solution should be installed
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/events.md]:
```suggestion
and aggregation capabilities, you can install a third-party solution
```

File: content/en/docs/concepts/cluster-administration/events.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+to capture events. Exploring third-party solutions is beyond the scope of the
+Kubernetes documentation. You may wish to look at the [CNCF Cloud Native Interactive Landscape](https://landscape.cncf.io)
+for projects that can consume and aggregate Kubernetes events.
+
+{{% /capture %}}
+
+{{% capture whatsnext %}}
+
+You can read more about using events for debugging your application in the [Application Introspection and Debugging](/docs/tasks/debug-application-cluster/debug-application-introspection/) section.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/events.md]:
```suggestion
* Read about using Events for debugging your application in the [Application Introspection and Debugging](/docs/tasks/debug-application-cluster/debug-application-introspection/) section.
```

File: content/en/docs/reference/glossary/events.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+---
+title: Events
+id: k8s-events
+date: 2019-08-30
+full_link: /docs/concepts/cluster-administration/events/
+short_description: >
+  Events are objects that provide insight into what is happening inside a {{< glossary_tooltip text="cluster" term_id="cluster" >}}.
```
Comment (sftim) [inline content/en/docs/reference/glossary/events.md]:
```suggestion
   Objects that record occurrences or actions inside your cluster.
```
[In my opinion], `short_description` fields work best without tooltips in them.

File: content/en/docs/reference/glossary/events.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+aka: 
+tags:
+- architecture
+- fundamental
+---
+ Kubernetes events are objects that provide insight into what is happening inside a {{< glossary_tooltip text="cluster " term_id="cluster" >}}, such as what decisions were made by the scheduler or why some pods were evicted from the node.
+    
+<!--more--> 
+
```
Comment (sftim) [inline content/en/docs/reference/glossary/events.md]:
```suggestion
Some resources use Events to broadcast information about errors, changes in their state, or other information that should be recorded and made available to cluster operators.
```
(not sure about this wording BTW)

**Comment by aimeeu** [comment]
Hi @thecrudge 
Yep, Fluentd is a CNCF project, but Elastic, while a member company, isn't actually a CNCF project listed in [Sandbox](https://www.cncf.io/sandbox-projects) or [Graduated/Incubating](https://www.cncf.io/projects/). BTW I'll be updating the Content Guide in the coming weeks, so I'll make sure to clarify CNCF member project vs projects from member companies.

If I understand the purpose of Fluentd correctly, it's not a logging agent but rather a layer between logging agent and Kubernetes. So I put the blurb in content/en/docs/concepts/cluster-administration/logging.md about using the CNCF landscape to look for logging agents.  :)

> Both Fluentd and Elastic are part of the CNCF. Other than that

File: content/en/docs/concepts/cluster-administration/events.md (hunk 7 above)
Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/events.md]: Aaargghh foiled again by the Style Guide!

File: content/en/docs/reference/glossary/events.md (hunk 1 above)
Comment (aimeeu) [reply content/en/docs/reference/glossary/events.md]: added

File: content/en/docs/reference/glossary/events.md (hunk 2 above)
Comment (aimeeu) [reply content/en/docs/reference/glossary/events.md]: Ooops - corrected!

File: content/en/docs/reference/glossary/events.md (hunk 3 above)
Comment (aimeeu) [reply content/en/docs/reference/glossary/events.md]: done

File: content/en/docs/concepts/cluster-administration/events.md (hunk 4 above)
Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/events.md]: done

File: content/en/docs/concepts/cluster-administration/events.md (hunk 5 above)
Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/events.md]: done

File: content/en/docs/concepts/cluster-administration/events.md (hunk 6 above)
Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/events.md]: done - thanks! Need to **really** brush up on my Kubernetes knowledge!

File: content/en/docs/concepts/cluster-administration/events.md (hunk 8 above)
Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/events.md]: done

File: content/en/docs/reference/glossary/events.md (hunk 9 above)
Comment (aimeeu) [reply content/en/docs/reference/glossary/events.md]: removed tooltip

File: content/en/docs/reference/glossary/events.md (hunk 10 above)
Comment (aimeeu) [reply content/en/docs/reference/glossary/events.md]: done!

**Comment by zacharysarah** [comment]
/hold

Pending discussion on #15748

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 11]
```diff
@@ -132,13 +159,28 @@ While Kubernetes does not provide a native solution for cluster-level logging, t
...
-
-Because the logging agent must run on every node, it's common to implement it as either a DaemonSet replica, a manifest pod, or a dedicated native process on the node. However the latter two approaches are deprecated and highly discouraged.
-
-Using a node-level logging agent is the most common and encouraged approach for a Kubernetes cluster, because it creates only one agent per node, and it doesn't require any changes to the applications running on the node. However, node-level logging _only works for applications' standard output and standard error_.
-
-Kubernetes doesn't specify a logging agent, but two optional logging agents are packaged with the Kubernetes release: [Stackdriver Logging](/docs/user-guide/logging/stackdriver) for use with Google Cloud Platform, and [Elasticsearch](/docs/user-guide/logging/elasticsearch). You can find more information and instructions in the dedicated documents. Both use [fluentd](http://www.fluentd.org/) with custom configuration as an agent on the node.
```
Comment (sftim) [inline content/en/docs/concepts/cluster-administration/logging.md]: Overall this is a useful set of changes. However, the content guidelines themselves are under discussion.

I wonder: would it be OK to leave in mentions of Elasticsearch, Stackdriver Logging, and fluentd, and then see how this PR looks?

Maybe reword the original text, maybe move the information to a What's Next? section. Also retain the relevant Task pages.

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 12]
```diff
@@ -9,9 +9,23 @@ weight: 60
...
+some kind of logging. The easiest and most embraced logging method for 
+containerized applications is to write to the standard output and standard 
+error streams.
+
+However, the native functionality provided by a container engine or runtime is 
+usually not enough for a complete logging solution. For example, if a container 
+crashes, a pod is evicted, or a node dies, you'll usually still want to access 
+your application's logs. As such, logs should have a separate storage and 
+lifecycle independent of nodes, pods, or containers. This concept is called 
+_cluster-level-logging_. Cluster-level logging requires a separate backend to
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/logging.md]: Nit:
```suggestion
_cluster-level logging_. Cluster-level logging requires a separate backend to 
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 13]
```diff
@@ -55,24 +69,39 @@ The output is:
...
-Everything a containerized application writes to `stdout` and `stderr` is handled and redirected somewhere by a container engine. For example, the Docker container engine redirects those two streams to [a logging driver](https://docs.docker.com/engine/admin/logging/overview), which is configured in Kubernetes to write to a file in json format.
+Everything a containerized application writes to `stdout` and `stderr` is 
+handled and redirected somewhere by a container engine. For example, the Docker 
+container engine redirects those two streams to [a logging 
+driver](https://docs.docker.com/engine/admin/logging/overview), which is 
+configured in Kubernetes to write to a file in json format.
 
 {{< note >}}
-The Docker json logging driver treats each line as a separate message. When using the Docker logging driver, there is no direct support for multi-line messages. You need to handle multi-line messages at the logging agent level or higher.
+The Docker json logging driver treats each line as a separate message. When
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/logging.md]: Nit:
```suggestion
The Docker JSON logging driver treats each line as a separate message. When 
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 14]
```diff
@@ -55,24 +69,39 @@ The output is:
...
 ## Logging at the node level
 
 ![Node level logging](/images/docs/user-guide/logging/logging-node-level.png)
 
-Everything a containerized application writes to `stdout` and `stderr` is handled and redirected somewhere by a container engine. For example, the Docker container engine redirects those two streams to [a logging driver](https://docs.docker.com/engine/admin/logging/overview), which is configured in Kubernetes to write to a file in json format.
+Everything a containerized application writes to `stdout` and `stderr` is 
+handled and redirected somewhere by a container engine. For example, the Docker 
+container engine redirects those two streams to [a logging 
+driver](https://docs.docker.com/engine/admin/logging/overview), which is 
+configured in Kubernetes to write to a file in json format.
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/logging.md]: Nit:
```suggestion
configured in Kubernetes to write to a file in JSON format.
```

File: content/en/docs/concepts/cluster-administration/logging.md [hunk 15]
```diff
@@ -132,13 +159,28 @@ While Kubernetes does not provide a native solution for cluster-level logging, t
...
+
+Using a node-level logging agent is the most common and encouraged approach for 
+a Kubernetes cluster, because it creates only one agent per node, and it 
+doesn't require any changes to the applications running on the node. However, 
+node-level logging _only works for applications' standard output and standard 
+error_.
+
+Kubernetes doesn't specify a logging agent, and no logging agents are
+packaged with the Kubernetes core release. Third-party logging agents are
+beyond the scope of the Kubernetes documentation. The [CNCF Cloud Native
```
Comment (zacharysarah) [inline content/en/docs/concepts/cluster-administration/logging.md]: Remove:
> Third-party logging agents are
beyond the scope of the Kubernetes documentation. 

It's true, but it's not immediately helpful in this context.

File: content/en/docs/concepts/cluster-administration/logging.md (hunk 11 above)
Comment (zacharysarah) [reply content/en/docs/concepts/cluster-administration/logging.md]: @sftim 
> I wonder: would it be OK to leave in mentions of Elasticsearch, Stackdriver Logging, and fluentd, and then see how this PR looks?

Given this update from @aimeeu:
> Kubernetes doesn't specify a logging agent, and no logging agents are
packaged with the Kubernetes core release.

All concerns about third-party content aside: if logging agents aren't packaged with the release anymore, then I think the proposed changes make sense. I also think it makes sense to get feedback from @piosz 👋 and/or @kubernetes/sig-instrumentation-bugs.

Comment (brancz) [reply content/en/docs/concepts/cluster-administration/logging.md]: Did we ever actually ship an agent with releases? I think everything we had were external to the kube release itself. I think going with a neutral statement here is the right way to go.

Comment (brancz) [reply content/en/docs/concepts/cluster-administration/logging.md]: cc @coffeepac

Comment (aimeeu) [reply content/en/docs/concepts/cluster-administration/logging.md]: @brancz  When I wrote this PR, I wasn't sure about logging agents being packaged with Kubernetes, so I asked in the  #sig-docs channel. Timothy St. Clair [replied](https://kubernetes.slack.com/archives/C1J0BPD2M/p1567521075006200), "Logging is external from core and there are EFK helm charts that are community supported.  That falls into the external add-ons category and will never be part of k8s release."   So I decided to point users to the CNCF landscape for logging agents rather than mentioning specific ones.

Comment (coffeepac) [reply content/en/docs/concepts/cluster-administration/logging.md]: Yep.  I'm an owner of the ES addon and removing the reference to this stuff sounds correct to me as well.

**Comment by sftim** [comment]
/close
It's fine to reopen this later
